Core/src/Surfaces/CylinderBounds.cpp

```diff
@@ -167,7 +167,12 @@ std::vector<Acts::Vector3> Acts::CylinderBounds::createCircles(
   if ((bevelMinZ != 0. || bevelMaxZ != 0.) && vertices.size() % 2 == 0) {
     auto halfWay = vertices.end() - vertices.size() / 2;
     double mult{1};
-    auto func = [&mult](Vector3& v) { v(2) += v(1) * mult; };
+    auto invCtrans = ctrans.inverse();
```

Comment:
Sorry to comment again, but I think if you use `auto` here, it won't actually do the inversion here (might be wrong). You can force this by either using a specific type or using `inverse().eval()`.

Reply:
Seems to work for me, at least dumping a cylinder (or even multiple) gives the expected result
![image](https://user-images.githubusercontent.com/94444689/158330634-5cffc435-e457-4ca8-8427-066817849d3c.png)